include/message_filters/sync_policies/approximate_epsilon_time.h

```diff
@@ -0,0 +1,357 @@
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+#ifndef MESSAGE_FILTERS__SYNC_POLICIES__APPROXIMATE_EPSILON_TIME_H_
+#define MESSAGE_FILTERS__SYNC_POLICIES__APPROXIMATE_EPSILON_TIME_H_
+
+#include <deque>
+#include <string>
+#include <tuple>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include "message_filters/connection.h"
+#include "message_filters/message_traits.h"
+#include "message_filters/null_types.h"
+#include "message_filters/signal9.h"
+#include "message_filters/synchronizer.h"
+
+namespace message_filters
+{
+namespace sync_policies
+{
+
+template<typename M0, typename M1, typename M2 = NullType, typename M3 = NullType, typename M4 = NullType,
+         typename M5 = NullType, typename M6 = NullType, typename M7 = NullType, typename M8 = NullType>
+class ApproximateEpsilonTime : public PolicyBase<M0, M1, M2, M3, M4, M5, M6, M7, M8>
+{
+public:
+  typedef Synchronizer<ApproximateEpsilonTime> Sync;
+  typedef PolicyBase<M0, M1, M2, M3, M4, M5, M6, M7, M8> Super;
+  typedef typename Super::Messages Messages;
+  typedef typename Super::Signal Signal;
+  typedef typename Super::Events Events;
+  typedef typename Super::RealTypeCount RealTypeCount;
+  typedef typename Super::M0Event M0Event;
+  typedef typename Super::M1Event M1Event;
+  typedef typename Super::M2Event M2Event;
+  typedef typename Super::M3Event M3Event;
+  typedef typename Super::M4Event M4Event;
+  typedef typename Super::M5Event M5Event;
+  typedef typename Super::M6Event M6Event;
+  typedef typename Super::M7Event M7Event;
+  typedef typename Super::M8Event M8Event;
+  typedef Events Tuple;
+
+  ApproximateEpsilonTime(uint32_t queue_size, rclcpp::Duration epsilon)
+  : parent_(nullptr)
+  , queue_size_(queue_size)
+  , epsilon_{epsilon}
+  {
+  }
+
+  ApproximateEpsilonTime(const ApproximateEpsilonTime& e)
+  : epsilon_{e.epsilon_}
+  {
+    *this = e;
+  }
+
+  ApproximateEpsilonTime& operator=(const ApproximateEpsilonTime& rhs)
+  {
+    parent_ = rhs.parent_;
+    queue_size_ = rhs.queue_size_;
+    events_ = rhs.events_;
+    epsilon_ = rhs.epsilon_;
+
+    return *this;
+  }
+
+  void initParent(Sync* parent)
+  {
+    parent_ = parent;
+  }
+
+  template<size_t i>
+  void add(const typename std::tuple_element<i, Events>::type& evt)
+  {
+    RCUTILS_ASSERT(parent_);
+
+    std::lock_guard<std::mutex> lock(mutex_);
+
+    // Tuple& t = tuples_[mt::TimeStamp<typename std::tuple_element<i, Messages>::type>::value(*evt.getMessage())];
+    auto & events_of_this_type = std::get<i>(events_);
+    if (0u == events_of_this_type.size()) {
+      ++number_of_non_empty_events_;
+    }
+    events_of_this_type.push_back(evt);
+    if (number_of_non_empty_events_ == RealTypeCount::value) {
+      process();
+    } else if (events_of_this_type.size() > queue_size_) {
+      // erase_old_events();
+      erase_begin_of_vector<i>();
```

Comment:
```suggestion
      erase_beginning_of_vector<i>();
```

Reply:
Fixed in https://github.com/ros2/message_filters/pull/84/commits/fbae9ae7e0bf103eb11e077071a15ef6e9dd0498